src/Search.elm

```diff
@@ -275,39 +287,20 @@ update path navKey msg model =
             )
 
 
+{-| TODO: Sort should be part of Route type
+-}
 createUrl :
-    String
+    SearchRoute
     -> String
     -> Maybe String
     -> Maybe String
     -> Int
     -> Int
     -> Sort
     -> String
-createUrl path channel query show from size sort =
-    [ Url.Builder.int "from" from
-    , Url.Builder.int "size" size
-    , Url.Builder.string "sort" <| toSortId sort
-    , Url.Builder.string "channel" channel
-    ]
-        |> List.append
-            (query
-                |> Maybe.map
-                    (\x ->
-                        [ Url.Builder.string "query" x ]
-                    )
-                |> Maybe.withDefault []
-            )
-        |> List.append
-            (show
-                |> Maybe.map
-                    (\x ->
-                        [ Url.Builder.string "show" x
-                        ]
-                    )
-                |> Maybe.withDefault []
-            )
-        |> Url.Builder.absolute [ path ]
+createUrl toRoute channel query show from size sort =
+    toRoute (Just channel) query show (Just from) (Just size) (Just <| toSortId sort)
+        |> Route.routeToString
```

Comment:
unification of URL building logic so we can keep fixes in single place.

Reply:
:1st_place_medal: